crates/sui-indexer-alt-framework/src/store.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+pub use crate::pipeline::sequential::Handler as SequentialHandler;
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use scoped_futures::ScopedBoxFuture;
+use std::time::Duration;
+
+pub use scoped_futures;
+
+/// Represents a database connection that can manage watermarks for pipeline operations in the
+/// framework.
+///
+/// This trait provides methods for the necessary reads and updates for the committer, reader, and
+/// pruner components of a data pipeline, allowing for tracking progress and coordination between
+/// different pipeline stages.
+#[async_trait]
+pub trait DbConnection: Send + Sync {
+    /// Given a pipeline, return the committer watermark from the database. This is used on indexer
+    /// startup to determine which checkpoint to start or continue processing from.
+    async fn committer_watermark(
+        &mut self,
+        pipeline: &'static str,
+    ) -> anyhow::Result<Option<CommitterWatermark>>;
+
+    /// Given a pipeline, return the reader watermark from the database. This is used by the indexer
+    /// to report progress to the database.
```

Comment:
nit: this is a getter, I think it's used by the pruner to calculate `reader_lo`.